l1-contracts/src/core/libraries/crypto/InboxAnchorChain.sol

```diff
@@ -0,0 +1,267 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.27;
+
+import {Hash} from "@aztec/core/libraries/crypto/Hash.sol";
+import {Errors} from "@aztec/core/libraries/Errors.sol";
+
+/**
+ * @notice    A hash of an InboxAnchor.
+ * @dev       This is a wrapper around bytes32 to make it harder to misuse.
+ */
+type InboxAnchorHash is bytes32;
+
+/**
+ * @notice    An anchor is a block that has a real value.
+ * @param root - The root of the tree for the block number
+ * @param blockNumber - the number of the block
+ * @param parentBlockNumber - The block number of the anchor parent
+ *                            used to make it simpler to find starting points
+ * @param parent - The hash of the anchor parent
+ */
+struct InboxAnchor {
+  bytes32 root;
+  uint256 blockNumber;
+  InboxAnchorHash parent;
+}
+
+struct InboxAnchorChain {
+  mapping(uint256 blockNumber => InboxAnchorHash link) links;
```

Comment:
How about circular storage here? Seems like we don't need to store historical values, right?

Reply:
It is one of those pesky "probably but a little more annoying". 

Right now it serves 2 purposes:
1. Simple to check against
2. Simple to build the anchor chain 

If we adopt the suggestion from earlier, and only need to check the end (which I think is good), we can make it circular, but with some caveats.

First, we would need to keep track of the anchor on the node to build the `anchorChain`, or have a very large circle. And secondly, because the links can be extended by people sending messages (without any consume calls happening) it might make the required `anchorChain` longer, which is a bit of a pain if moving them into the circuirts where it is very good to have guarantees about the length 👀.